pkg/controlplane/reconcilers/lease.go

```diff
@@ -132,9 +132,9 @@ func (s *storageLeases) Destroy() {
 
 // NewLeases creates a new etcd-based Leases implementation.
 func NewLeases(config *storagebackend.ConfigForResource, baseKey string, leaseTime time.Duration) (Leases, error) {
-	// note that newFunc, newListFunc and resourcePrefix
+	// note that newFunc, newListFunc
 	// can be left blank unless the storage.Watch method is used
-	leaseStorage, destroyFn, err := storagefactory.Create(*config, nil, nil, "")
+	leaseStorage, destroyFn, err := storagefactory.Create(*config, nil, nil, baseKey)
```

Comment:
if we pass this in as the prefix to the storage, don't we have to *stop* storing it in storageLeases and doing `path.Join(s.baseKey, ...)` in this file? same comment on peer_endpoint_lease

Reply:
No, we construct storage per resource, however technically storage can request any key from any other resource. `resourcePrefix` is a pretty new addition to storage. 

I think we should restrict this, and that's what I'm proposing in https://github.com/kubernetes/kubernetes/pull/133871

Reply:
> I think we should restrict this, and that's what I'm proposing in #133871

ah